flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java

```diff
@@ -44,6 +47,12 @@ public TestStreamEnvironment(
 				null);
 
 		setParallelism(parallelism);
+
+		if (Randomization) {
+			final String testName = TestNameProvider.getCurrentTestName();
```

Comment:
This seems a bit...janky? Why can we not mutate the configuration within the MiniClusterResource?

Reply:
I'm missing a hook where I can mutate the configuration per test method. `MiniClusterResource` is mostly used on class level. Any guidance is greatly appreciated.

Reply:
I see. I can't come up with a good alternative unfortunately.

Reply:
I moved it a bit outside into the factory, but that's the best that I came up with.